### ISSUE TITLE
Interoperable Data Model

### ISSUE THREAD

**Issue by bradley-holt** [ibm-watson-data-lab/shopping-list#81]
An interoperable data model has been outlined here:
https://github.com/ibm-watson-data-lab/shopping-list#interoperability

Please provide feedback on this design as comments on this ticket.

**Comment by vabarbosa** [comment]
would it be useful to include some sort of `count` or `size` to the [Shopping List](https://github.com/ibm-watson-data-lab/shopping-list#shopping-list-example) which indicates the number of items in the shopping list?

**Comment by bradley-holt** [comment]
@vabarbosa Good suggestion! I'm inclined to do this using a `_count` reduce function, rather than including it as metadata within a shopping list (which I can add to the README). This way the shopping list document does not need to be updated every time that a shopping list item is added or removed from the shopping list. What are your thoughts on that?

**Comment by vabarbosa** [comment]
@bradley-holt works for me. i started thinking the same shortly after i submitted the comment.

**Comment by bradley-holt** [comment]
I've made several changes to the data model to account for only using Mango and not map/reduce for queries. The reason for this is that Cloudant Sync does not support map/reduce and also that it is best to avoid map/reduce in PouchDB. The least common denominator here is Mango.

Specific changes include:
* The corresponding map feature is now referenced from shopping lists. This is for several reasons, but mainly because I am no longer relying on the linked documents feature of CouchDB and its variants (the example for this has been deleted). This also allows geolocation context documents to be deleted when they are no longer needed, while still preserving the link between a shopping list and a map feature.
* The example of querying a shopping list and all of its list items together now uses Mango.
* A note was added on how to get a count of items within a shopping list (/cc @vabarbosa).
* The example of querying for a geolocation context near the current geolocation coordinates of the device now uses Mango.
